ibm_db.c

```diff
@@ -17161,6 +17057,7 @@ static PyObject *ibm_db_fetchall(PyObject *self, PyObject *args)
         return NULL;
     }
     LogMsg(DEBUG, "Initialized result list", fileName);
+    Py_XDECREF(return_value);
```

Comment:
@bchoudhary6415  return_value is already NULL here, so calling `Py_XDECREF(return_value);` not required here. Remove this line.